Use _process_group in ReplicatedTensor and ShardedTensor.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #77209
* #77191

As per comment in
https://github.com/pytorch/pytorch/pull/77191#discussion_r869641344, making
everthing consistent amongst these tensors.

Differential Revision: [D36300798](https://our.internmc.facebook.com/intern/diff/D36300798/)

pritamdamania87 committed:
Use _process_group in ReplicatedTensor and ShardedTensor.

As per comment in
https://github.com/pytorch/pytorch/pull/77191#discussion_r869641344, making
everthing consistent amongst these tensors.

Differential Revision: [D36300798](https://our.internmc.facebook.com/intern/diff/D36300798/)

[ghstack-poisoned]

wanchaol commented on torch/distributed/_shard/replicated_tensor.py:
```diff
@@ -33,15 +33,15 @@ class ReplicatedTensor(torch.Tensor):
-    process_group: distributed_c10d.ProcessGroup
+    _process_group: distributed_c10d.ProcessGroup
 
     __slots__ = ["process_group"]
```
nit: fix this `__slots__`?

wanchaol reviewed (approved):
looks good, thanks for fixing this!

fduwjj reviewed (approved):
LGTM

pritamdamania87 committed:
Update on "Use _process_group in ReplicatedTensor and ShardedTensor."


As per comment in
https://github.com/pytorch/pytorch/pull/77191#discussion_r869641344, making
everthing consistent amongst these tensors.

Differential Revision: [D36300798](https://our.internmc.facebook.com/intern/diff/D36300798/)

[ghstack-poisoned]

pritamdamania87 commented:
@pytorchbot merge this please
